lib/puppet/util/windows/user.rb

```diff
@@ -95,8 +98,16 @@ def logon_user(name, password, &block)
     # token has been closed by this point
     true
   end
+
   module_function :logon_user
 
+  def logon_user_by_logon_type(name, password, logon_type, logon_provider, token)
+    LogonUserW(wide_string(name), wide_string('.'), password.nil? ? FFI::Pointer::NULL : wide_string(password), logon_type, logon_provider, token)
+  end
+
+  module_function :logon_user_by_logon_type
```

Comment:
Same comment here about removing `module_function`. I don't think we want to expose this method publicly.

Reply:
I'm not s

Reply:
@Iristyle there is a private_class_method bellow the module_function.
If I remove the module_function it will fail to run because it will not find the method logon_user_by_logon_type.

Reply:
@Iristyle removed module_function from there, added self.logon_user_by_logon_type as the definition and added private_class_method